From 3b1a5c4f238b0b9e691a7e32bda2efd577b9e675 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Mon, 20 Jan 2025 11:12:24 +0100
Subject: [PATCH 1522/1695] dts: stm32: remove OPP 400 on stm32mp15x

Remove OPP 400 MHz on stm32mp15x.
Make OPP 650 MHz compatible with all SOCs variants.
Boot at nominal 650 MHz frequency and to not use overdrive OPP in case of
DVFS is not enabled in linux.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e6f7f91a456e8b91a3896e00288d7a796e3341c3 dts: stm32: remove OPP 400 on stm32mp15x]
Change-Id: I7ed07c9ed4211eb5ee72d340d52c264a11c17fd9
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/429933
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/451432
Tested-by: Pascal PAILLET <p.paillet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp151.dtsi | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index e35326445..877035d05 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -40,17 +40,10 @@
 		compatible = "operating-points-v2";
 		opp-shared;
 
-		opp-400000000 {
-			opp-hz = /bits/ 64 <400000000>;
-			opp-microvolt = <1200000>;
-			opp-supported-hw = <0x2>;
-			opp-suspend;
-		};
-
 		opp-650000000 {
 			opp-hz = /bits/ 64 <650000000>;
 			opp-microvolt = <1200000>;
-			opp-supported-hw = <0x1>;
+			opp-supported-hw = <0x3>;
 			st,opp-default;
 		};
 
@@ -58,7 +51,6 @@
 			opp-hz = /bits/ 64 <800000000>;
 			opp-microvolt = <1350000>;
 			opp-supported-hw = <0x2>;
-			st,opp-default;
 		};
 	};
 
-- 
2.47.2

